docs/project/requirements/architecture.md

```diff
@@ -0,0 +1,147 @@
+# Architecture
+
+Un schéma d'implémentation détaillé sera établi pour décrire l'architecture globale de l'infrastructure, y compris les différents composants, leur interaction et leur déploiement.
+
+
+## Application
+
+
+### Proxy: Traefik
+
+* Application router
+* High-availability by fault tolerance
+* Load-balancing by requests distribution
+  ALB ?
+* Helm chart
+* Version
+
+
+### Application: FastAPI
+
+* Python
+  * Version:
+  * Packaging
+* Project management? Poetry?
+
+
+### Databases: PostgreSQL
+
+* Helm chart
+* Version
+* Master/slave for high availability and read load balancing
+* Master failover
+
+
+## Infrastructure
+
+
+### Network
+
+
+#### Application Load Balancer (ALB)
+
+
+#### Virtual Private Cloud (VPC)
+
+* VPC subnet
+
+
+#### Internet Gateways (IGW)
+
+* Global egress
+
+
+#### Availability Zones (AZ)
+
+* 2 availability zones for high availability
+
+
+#### NAT Gateways (NGW)
+
+* Per AZ egress
+* In public subnet
+
+
+#### Subnets
+
+
+##### Public subnets
+
+* Application access
+* Bastion access
+* NAT gateways ?
+
+
+##### Private subnets
+
+* Database
+* Admin
+
+
+### Compute
+
+#### Nodes
+
+* EC2 compute nodes
+  * Application nodes
+  * Database nodes
+  * Bastion node
+* Auto-scaling groups
+
+
+#### Orchestration: Kubernetes
+
+#### Pods
+
+##### ReplicaSets
+
+* Traefik
+* Application
+* Database
+* cert-manager
+* Metrics server
+* bastion
+
+##### DaemonSets
+
+* Any ?
+
+
+### Storage
+
+#### CSI
+
+##### EBS
+
+* FIXME
```

Comment:
We're using the aws-ebs-csi-driver cluster addon from EKS. This registers EBS as a backend for Persistant Volumes.

Reply:
Fixed